### PR TITLE
Make possible to separate hw offline wallet leafs like for origin hw wo wallet

### PR DESCRIPTION
**Pull request by romaniy173**
Issue: fix trading and create tx mechanism to make possible to separate offline hw wallet per purpose.
http://185.213.153.35:8081/browse/BST-2681

Note: related on https://github.com/BlockSettle/common/pull/1920

### CODE REVIEW

File: BlockSettleHW/ledger/ledgerDevice.cpp [hunk 1]
```diff
@@ -479,9 +479,10 @@ void LedgerCommandThread::processGetPublicKey()
    else {
       logger_->debug(
          "[LedgerCommandThread] getPublicKey - Operation succeeded.\nRoot xpub : "
-         + walletInfo.info_.xpubRoot_ + " \nNested xpub: "
-         + walletInfo.info_.xpubNestedSegwit_ + " \nNativeSegwit: "
-         + walletInfo.info_.xpubNativeSegwit_);
+         + walletInfo.info_.xpubRoot + " \nNested xpub: "
+         + walletInfo.info_.xpubNestedSegwit + " \nNativeSegwit: "
+         + walletInfo.info_.xpubNativeSegwit + "\nLegacy: "
+         + walletInfo.info_.xpubLegacy);
```
Comment (sergey-chernikov) [inline BlockSettleHW/ledger/ledgerDevice.cpp:485]: Using native fmt::format formatting will make it much more clear for the reader, what exactly is printed here. Also, putting a space (' ') before `\n` is pointless.

File: BlockSettleUILib/ChatUI/OTCShieldWidgets/OTCNegotiationRequestWidget.cpp [hunk 2]
```diff
@@ -283,7 +283,7 @@ void OTCNegotiationRequestWidget::onParentAboutToHide()
 void OTCNegotiationRequestWidget::onCurrentWalletChanged()
 {
    auto recvHdWallet = getCurrentHDWallet();
-   if (recvHdWallet->isHardwareWallet()) {
+   if (!recvHdWallet->canMixLeafs()) {
```
Comment (sergey-chernikov) [inline BlockSettleUILib/ChatUI/OTCShieldWidgets/OTCNegotiationRequestWidget.cpp:286]: Multiple for "leaf" is "leaves", not "leafs".

File: BlockSettleHW/ledger/ledgerDevice.cpp (hunk 1 above)
Comment (romaniy173) [reply BlockSettleHW/ledger/ledgerDevice.cpp]: done

File: BlockSettleUILib/ChatUI/OTCShieldWidgets/OTCNegotiationRequestWidget.cpp (hunk 2 above)
Comment (romaniy173) [reply BlockSettleUILib/ChatUI/OTCShieldWidgets/OTCNegotiationRequestWidget.cpp]: Done, thanks

**Pull request merged**